automatic/sublimetext4/tools/chocolateyInstall.ps1

```diff
@@ -1,5 +1,7 @@
 ﻿$ErrorActionPreference = 'Stop'
 
+$softwareName = 'Sublime Text*'
```

Comment:
Hmm, I feel like we shouldn't be matching Sublime Text 3 here, considering that it's a separate package on Chocolatey.
It also appears to be possible to run them side-by-side:
https://www.sublimetext.com/docs/side_by_side.html

I'm not sure if we can phrase this with a single `-like` pattern though. Perhaps we should disable auto-uninstall altogether with `.skipAutoUninstall` file and just handle uninstall in `chocolateyUninstall.ps1` by trying 2 different registry key patterns - "Sublime Text 4" and "Sublime Text_is1" - without any wildcards.

Reply:
I wish I knew at what point/what ST4 version Sublime HQ Pty Ltd did this (re: changing the string from `Sublime Text 4` to `Sublime Text`), since it may very well be that nobody has the non-4 string in use any more, at least via Chocolatey.

The `_is1` suffix appears statically defined, according to [this reddit post](https://stackoverflow.com/questions/35156309/remove-is1-suffix-from-inno-setup-appid), which references the Inno Setup installer code.

`-like` has basic wildcard and range support, but no exclusion support (unlike `-match` which is regex-based).  Docs: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_wildcards?view=powershell-7.3

In short: probably need to check both registry keys without any wildcards, as as you said.

Reply:
I'm pretty sure I tested the uninstall script when I made the package so the old name should be present in the very first version pushed to chocolatey, at the very least. If you want to try it out then that's the version I would suggest to check.

Reply:
It just dawned on me: do you happen to know what `softwareName` even does on an **install**?  On an uninstall I understand the relevance, but not on an install.  The Chocolatey docs<sup>[[1]](https://docs.chocolatey.org/en-us/create/functions/install-chocolateypackage)</sup><sup>[[2]](https://docs.chocolatey.org/en-us/create/functions/install-chocolateyinstallpackage)</sup> don't explain it, and [the code](https://github.com/chocolatey/choco/blob/master/src/chocolatey.resources/helpers/functions/Install-ChocolateyInstallPackage.ps1) doesn't even reference it.

Reply:
I assumed that it's used by auto uninstaller but it either isn't the case or is only the case in some edge cases since auto uninstaller uninstalls Sublime Text fine despite incorrect pattern being used.

Reply:
I found https://github.com/pauby/ChocoPackages/blob/master/automatic/webview2-runtime/tools/chocolateyinstall.ps1 which has the comment:

> #part or all of the Display Name as you see it in Programs and Features. It should be enough to be unique

So I think for the **install**, just saying `Sublime Text` is enough, since at least in Build 4143, that is the literal string that shows up for me in Programs and Features/list of programs to uninstall.  There is no version or build string or architecture (x86/x64) at the end.

Ironically, the uninstaller script for the same package -- https://github.com/pauby/ChocoPackages/blob/master/automatic/webview2-runtime/tools/chocolateyuninstall.ps1 -- does not use a wildcard at all.  This is just another example of the incongruities I find throughout Chocolatey stuff, even when the package maintainer is a Chocolatey engineering manager.  :)

In short: I'm not sure we need to do a double-registry-check at all.  Say people are using sublimetext4 Choco package 4.0.0.1234 (just making up a number here).  If they do a `choco upgrade sublimetext4`, this will get them 4.0.0.414300.  The chocoUninstall.ps1 that runs, prior to the upgrade, would be the one for 4.0.0.1234, which **would not** contain the changes we're talking about here.  It would be the one from whenever you packed + pushed 4.0.0.1234.

Reply:
Ah yes! We definitely want `_is1` here, I forgot to mention that.

As for `softwareName` in the installer: yep, figured as much (looks like the parameter ends up in the "ignored params" section).  Great, I'll clean that up too.

Reply:
Discussed further below, resolving this convo.